docker-java-transport-okhttp/src/main/java/com/github/dockerjava/okhttp/UnixDomainSocket.java

```diff
@@ -232,20 +232,12 @@ public void shutdownOutput() {
         public int read(byte[] bytesEntry, int off, int len) throws IOException {
             try {
                 if (off > 0) {
-                    int bytes = 0;
-                    int remainingLength = len;
-                    int size;
                     byte[] data = new byte[(len < 10240) ? len : 10240];
-                    do {
-                        size = recv(fd, data, (remainingLength < 10240) ? remainingLength : 10240, 0);
-                        if (size > 0) {
-                            System.arraycopy(data, 0, bytesEntry, off, size);
-                            bytes += size;
-                            off += size;
-                            remainingLength -= size;
-                        }
-                    } while ((remainingLength > 0) && (size > 0));
-                    return bytes;
+                    int size = recv(fd, data, (len < 10240) ? len : 10240, 0);
```

Comment:
the loop was there in case `len > 10240`, but now the code seems to be truncating

Reply:
Sorry I thought I responded to the comment but I managed to do smth else 😬

So this is true that now if `len` is greater than `10240` we tried to `recv()` from socket only `10240` bytes. But it is totally fine because it does not violate the contract of `read()` method:
> Reads up to len bytes of data from the input stream into an array of bytes. An attempt is made to read as many as len bytes, but a smaller number may be read. The number of bytes actually read is returned as an integer.

On the contrary, the attempt to call `recv()` several times may end with the blocking call if the amount of data available in the socket is less or equal to `10240`. Because following the manual of `recv()` system call:
> If no messages are available at the socket, the receive calls wait for a message to arrive, unless the socket is nonblocking (see fcntl(2)), in which case the value -1 is returned and the external variable errno is set to EAGAIN or EWOULDBLOCK.

In a nutshell, the single call of `recv(20480)` is not equal to two calls of `recv(10240)` because the second call might become blocking and the program will be stuck at this point waiting for data.